nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBDirectory.java

```diff
@@ -93,14 +115,47 @@ public Session getSession() throws DirectoryException {
             session.getCollection(countersCollectionName).insertOne(MongoDBSerializationHelper.fieldMapToBson(seq));
         }
 
-        if (!initialized && descriptor.getDataFileName() != null && !session.hasCollection(getName())) {
-            DirectoryCSVLoader.loadData(descriptor.getDataFileName(), descriptor.getDataFileCharacterSeparator(),
-                    schema, session::createEntry);
+        if (!initialized) {
+            String policy = descriptor.getCreateTablePolicy();
+            switch (policy) {
+            case CREATE_TABLE_POLICY_ALWAYS:
+                session.getCollection(getName()).drop();
+                loadData(schema, session);
+                break;
+            case CREATE_TABLE_POLICY_ON_MISSING_COLUMNS:
+                if (session.hasCollection(getName())) {
+                    boolean missingColumns = schema.getFields()
+                                                .stream()
+                                                .anyMatch(f -> session.getCollection(getName())
```

Comment:
To improve performance you should :
- refactor to a local variable `session.getCollection(getName())` outside of the stream.
- use `count()` method instead of `find()` ?
To improve readability, maybe use `map()` stream method to get name and then local name from fields.